qiskit_experiments/data_processing/nodes.py

```diff
@@ -0,0 +1,244 @@
+
+    def __init__(self, scale: Optional[float] = 1.0, average: bool = False):
+        """
+        Args:
+            scale: scale by which to multiply the real part of the data.
+            average: if True the single-shots are averaged.
+        """
+        self.scale = scale
+        self.average = average
+        super().__init__()
+
+    @property
+    def node_output(self) -> str:
+        """Key under which ToReal stores the data."""
+        return "memory_real"
+
+    @property
+    def node_inputs(self) -> List[str]:
+        """Returns a list of input data that the node can process."""
+        return ["memory"]
+
+    def process(self, data: Dict[str, Any]):
+        """
+        Modifies the data inplace by taking the real part of the memory and
+        scaling it by the given factor.
+
+        Args:
+            data: The data dict. IQ data is stored under memory.
+
+        Raises:
+            DataProcessorError: if the data does not contain memory.
+        """
+
+        # Single shot data
+        if isinstance(data["memory"][0][0], list):
+            new_mem = []
+            for shot in data["memory"]:
+                new_mem.append([self.scale * _[0] for _ in shot])
+
+            if self.average:
+                new_mem = list(np.mean(np.array(new_mem), axis=0))
+
+        # Averaged data
+        else:
+            new_mem = [self.scale * _[0] for _ in data["memory"]]
+
+        data[self.node_output] = new_mem
+
+
+@iq_data
+@prev_node(Kernel)
+class ToImag(DataAction):
+    """IQ data post-processing. This returns imaginary part of IQ data."""
+
+    def __init__(self, scale: Optional[float] = 1.0, average: bool = False):
+        """
+        Args:
+            scale: scale by which to multiply the imaginary part of the data.
+        """
+        self.scale = scale
+        self.average = average
+        super().__init__()
+
+    @property
+    def node_output(self) -> str:
+        """Key under which ToImag stores the data."""
+        return "memory_imag"
+
+    @property
+    def node_inputs(self) -> List[str]:
+        """Returns a list of input data that the node can process."""
+        return ["memory"]
+
+    def process(self, data: Dict[str, Any]):
+        """
+        Scales the imaginary part of IQ data.
+
+        Args:
+            data: The data dict. IQ data is stored under memory.
+
+        Raises:
+            DataProcessorError: if the data does not contain memory.
+        """
+
+        # Single shot data
+        if isinstance(data["memory"][0][0], list):
+            new_mem = []
+            for shot in data["memory"]:
+                new_mem.append([self.scale * _[1] for _ in shot])
+
+            if self.average:
+                new_mem = list(np.mean(np.array(new_mem), axis=0))
+
+        # Averaged data
+        else:
+            new_mem = [self.scale * _[0] for _ in data["memory"]]
+
+        data[self.node_output] = new_mem
+
```

Comment:
I guess that `ToReal` and `ToImag` can be unified to avoid code duplication

Reply:
Done in 38d6675